### PR TITLE
[FIX] sale_order_type_automation: commit invoice creation and complet…

### PR DESCRIPTION
**Pull request by augusto-weiss**
…e rollback

### CODE REVIEW

**Comment by roboadhoc** [comment]
[Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/sale/pull/1018).

File: sale_order_type_automation/models/sale_order.py [hunk 1]
```diff
@@ -31,13 +31,15 @@ def run_invoicing_atomation(self):
             invoices = self._create_invoices(final=True)
             if not invoices:
                 continue
-            validate_using_try_except = bool(self._context.get("validate_using_try_except"))
+            validate_using_try_except = bool(self._context.get("validate_using_try_except")) or rec.type_id.invoicing_atomation == 'try_validate_invoice'
```
Comment (zaoral) [inline sale_order_type_automation/models/sale_order.py]:
```python
            if rec.type_id.invoicing_atomation in ['try_validate_invoice', 'validate_invoice']:
                if self._context.get("validate_using_try_except"):
                    rec.env.cr.commit()
                try:
                    invoices.sudo().action_post()
                except Exception as error:
                    rec.env.cr.rollback()
                    if no tself._context.get("validate_using_try_except"):
                         raise error
                    message = _(
                        "We couldn't validate the automatically created "
                        "invoices (ids %s), you will need to validate them"
                        " manually. This is what we get: %s") % (
                            invoices.ids, error)
                    invoices.message_post(body=message)
                    rec.message_post(body=message)
```

Comment (zaoral) [reply sale_order_type_automation/models/sale_order.py]: raise_invoice_validation_error

File: sale_order_type_automation/models/sale_order.py [hunk 2]
```diff
@@ -31,18 +31,20 @@ def run_invoicing_atomation(self):
...
                     invoices.sudo().action_post()
                 except Exception as error:
+                    rec.env.cr.rollback()
                     message = _(
                         "We couldn't validate the automatically created "
                         "invoices (ids %s), you will need to validate them"
                         " manually. This is what we get: %s") % (
                             invoices.ids, error)
+                    if not self._context.get("commit_invoice_automation"):
+                        raise UserError(message)
```
Comment (zaoral) [inline sale_order_type_automation/models/sale_order.py]:
```suggestion
                        raise error
```

**Review by zaoral — CHANGES_REQUESTED** [review]

**Review by zaoral — APPROVED** [review]

**Comment by zaoral** [comment]
@roboadhoc r+

**Pull request closed without merging**